### PR TITLE
Implement GridReactivePowerFormula

### LINKED ISSUE #1075: Implement a method to stream reactive power at the grid connection point
### What's needed?

A new stream `microgrid.grid().reactive_power` method.  This would need a new formula generator as well: `GridReactivePowerFormula`

### Proposed solution

_No response_

### Use cases

_No response_

### Alternatives and workarounds

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ela-kotulska-frequenz**

### CODE REVIEW

File: src/frequenz/sdk/timeseries/formula_engine/_formula_generators/_grid_power_formula.py [hunk 1]
```diff
@@ -32,3 +35,47 @@ def generate(  # noqa: DOC502
             Power.from_watts,
         )
         return self._generate(builder)
+
+    def _get_fallback_formulas(
+        self, components: set[Component]
+    ) -> dict[Component, FallbackFormulaMetricFetcher[Power] | None]:
+        """Find primary and fallback components and create fallback formulas.
+
+        The primary component is the one that will be used to calculate the producer power.
```
Comment (shsms) [inline src/frequenz/sdk/timeseries/formula_engine/_formula_generators/_grid_power_formula.py]:
```suggestion
        The primary component is the one that will be used to calculate the grid power.

File: src/frequenz/sdk/timeseries/formula_engine/_formula_generators/_grid_reactive_power_formula.py [hunk 2]
```diff
@@ -27,8 +30,52 @@ def generate(  # noqa: DOC502
...
+            ReactivePower.from_volt_amperes_reactive,
         )
         return self._generate(builder)
+
+    def _get_fallback_formulas(
+        self, components: set[Component]
+    ) -> dict[Component, FallbackFormulaMetricFetcher[ReactivePower] | None]:
+        """Find primary and fallback components and create fallback formulas.
+
+        The primary component is the one that will be used to calculate the producer power.
```
Comment (shsms) [inline src/frequenz/sdk/timeseries/formula_engine/_formula_generators/_grid_reactive_power_formula.py]:
```suggestion
        The primary component is the one that will be used to calculate the grid reactive power.

**Review by shsms — COMMENTED** [review]
Just a couple of documentation issues. Looks great otherwise!

File: src/frequenz/sdk/timeseries/formula_engine/_formula_generators/_grid_power_formula.py (hunk 1 above)
Comment (ela-kotulska-frequenz) [reply src/frequenz/sdk/timeseries/formula_engine/_formula_generators/_grid_power_formula.py]: Done

File: src/frequenz/sdk/timeseries/formula_engine/_formula_generators/_grid_reactive_power_formula.py (hunk 2 above)
Comment (ela-kotulska-frequenz) [reply src/frequenz/sdk/timeseries/formula_engine/_formula_generators/_grid_reactive_power_formula.py]: Done

**Review by shsms — APPROVED** [review]

**Pull request merged**